### PR TITLE
Update MapboxDirections to 2.4.0

### PR DESCRIPTION
**Pull request by azarovalex**
Also pinning Carthage and SPM deps to the latest available

### CODE REVIEW

File: Cartfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
 binary "https://api.mapbox.com/downloads/v2/carthage/mapbox-common/MapboxCommon.json" == 21.2.0
 binary "https://api.mapbox.com/downloads/v2/carthage/mobile-navigation-native/MapboxNavigationNative.xcframework.json" ~> 94.0
-github "mapbox/mapbox-directions-swift" == 2.4.0-rc.2
+github "mapbox/mapbox-directions-swift" ~> 2.4.0
```
Comment (bamx23) [inline Cartfile:3]: In `2.3.0` we used `~> 2.3` to allow "up to major update" here

File: Tests/CocoaPodsTest/PodInstall/Podfile.lock [hunk 2]
```diff
@@ -52,19 +52,19 @@ EXTERNAL SOURCES:
...
+  MapboxNavigationNative: 145551b74e5dab196b6b11adc366d9dce529c5d3
   MapboxSpeech: e4ed02984444b6373374c72c369edaf045cc490c
   Polyline: fce41d72e1146c41c6d081f7656827226f643dff
   Solar-dev: 4612dc9878b9fed2667d23b327f1d4e54e16e8d0
-  Turf: d8d52444483c968cf2d65b8b54640f8faf4dda03
+  Turf: 60b93cfdc62758526bc7bf1ef191a829aeb9694d
 
 PODFILE CHECKSUM: ce6227c2aeeca0d51c521648629d4f06ae38a439
 
-COCOAPODS: 1.11.3
```
Comment (bamx23) [inline Tests/CocoaPodsTest/PodInstall/Podfile.lock:70]: Could you update your CocoaPods?

File: Tests/CocoaPodsTest/PodInstall/Podfile.lock [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
-  - MapboxDirections-pre (2.4.0-rc.2):
+  - MapboxDirections (2.4.0):
     - Polyline (~> 5.0)
     - Turf (~> 2.0)
-  - MapboxMaps (10.4.1):
-    - MapboxCommon (= 21.2.0)
-    - MapboxCoreMaps (= 10.4.1)
+  - MapboxMaps (10.4.3):
+    - MapboxCommon (= 21.2.1)
+    - MapboxCoreMaps (= 10.4.2)
```
Comment (bamx23) [inline Tests/CocoaPodsTest/PodInstall/Podfile.lock:14]: Could you update resolved files for Maps in SPM too?

**Review by bamx23 — APPROVED** [review]

File: .circleci/config.yml [hunk 4]
```diff
@@ -154,7 +154,7 @@ commands:
             LOG=$(./scripts/swift-api-diff/.build/release/swift-api-diff breaking_changes << parameters.base_api_path >> << parameters.new_api_path >> --ignore-undocumented true)
             if [ ! -z "$CIRCLE_PULL_REQUEST" ]; then
               if [ -z "$LOG" ]; then
-                gh pr comment -b "No breaking changes detected in << parameters.module_name >>"
+                echo "No breaking changes detected in << parameters.module_name >>"
```
Comment (azarovalex) [inline .circleci/config.yml:157]: This is a small fix in main branch, backporting it to the release-v2.4 branch as well

**Pull request merged**